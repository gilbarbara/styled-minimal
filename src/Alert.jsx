import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

const styles = (props: Object): string => {
  const { size } = props;
  const { borderRadius, maxWidth, padding } = themeGet(props, 'alert');

  return css`
    ${baseStyles.variant};
    border-radius: ${px(borderRadius)};
    font-size: ${baseStyles.fontSize};
    line-height: ${baseStyles.lineHeight};
    max-width: ${px(maxWidth)};
    padding: ${px(padding[size][0])} ${px(padding[size][1])};
    width: 100%;

    a {
      ${baseStyles.color};
    }
  `;
};

const Alert = styled(Box).attrs({
  role: 'alert',
})`
  ${styles};
`;

Alert.displayName = 'Alert';

Alert.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  /** button size */
  size: sizesAllPropTypes,
  /** button variant */
  variant: variantPropTypes,
  ...Box.propTypes,
};

Alert.defaultProps = {
  outline: false,
  size: 'md',
  variant: 'primary',
};

export default Alert;
