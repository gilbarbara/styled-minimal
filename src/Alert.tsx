import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import {
  baseStyles,
  colorStyles,
  lineHeightStyles,
  sx,
  system,
  variantStyles,
} from './utils/system';

import { AlertProps } from './types';

export const StyledAlert = styled('div')<AlertProps>(
  props => {
    const { borderRadius, maxWidth, padding } = getTheme(props, 'alert');

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${px(borderRadius)};
      font-size: inherit;
      line-height: ${lineHeightStyles};
      max-width: ${px(maxWidth)};
      padding: ${padding};
      width: 100%;

      a {
        ${colorStyles};
      }
    `;
  },
  sx,
  system,
);

export const Alert = React.forwardRef<HTMLDivElement, AlertProps>((props, ref) => (
  <StyledAlert ref={ref} {...props} />
));

Alert.defaultProps = {
  dark: false,
  invert: false,
  role: 'alert',
  variant: 'primary',
};
