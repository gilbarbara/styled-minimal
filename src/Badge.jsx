import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, themeGet } from './utils/helpers';
import { baseStyles, sizesAllPropTypes, variantPropTypes } from './utils/system';

import Box from './Box';

export const Badge = styled(Box)`
  ${(props: Object): string => {
    const {
      borderRadius,
      fontSize,
      fontWeight,
      padding,
    } = themeGet(props, 'badge');
    const fontSizeProp = themeGet(props, 'componentSizes', { key: 'size' });

    return css`
      ${baseStyles.variant};
      border-radius: ${px(borderRadius)};
      display: inline-flex;
      font-size: ${fontSize};
      ${fontSizeProp ? `font-size: ${fontSizeProp}` : ''};
      font-weight: ${fontWeight};
      line-height: 1;
      padding: ${px(padding[0])} ${px(padding[1])};
      vertical-align: baseline;
    `;
  }};
`;

Badge.displayName = 'Badge';

Badge.propTypes = {
  as: PropTypes.string,
  children: PropTypes.node.isRequired,
  dark: PropTypes.bool,
  outline: PropTypes.bool,
  size: sizesAllPropTypes,
  variant: variantPropTypes,
  ...Box.propTypes,
};

Badge.defaultProps = {
  as: 'span',
};

export default Badge;
