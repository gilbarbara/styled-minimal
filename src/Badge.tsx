import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px } from './utils/helpers';
import { baseStyles, sx, system, variantStyles } from './utils/system';

import { BadgeProps } from './types';

export const StyledBadge = styled('sup')<BadgeProps>(
  props => {
    const { dot } = props;
    const { borderRadius, fontSize, fontWeight, lineHeight, padding, size, sizeDot } = getTheme(
      props,
      'badge',
    );

    return css`
      ${baseStyles};
      ${variantStyles};
      border-radius: ${dot ? '100%' : px(borderRadius)};
      display: inline-flex;
      font-size: ${dot ? 0 : px(fontSize)};
      font-weight: ${fontWeight};
      justify-content: center;
      line-height: ${dot ? 0 : px(lineHeight)};
      height: ${px(dot ? sizeDot : size)};
      padding: ${dot ? 0 : `0 ${px(padding)}`};
      min-width: ${px(dot ? sizeDot : size)};
    `;
  },
  sx,
  system,
);

export const Badge = React.forwardRef<HTMLSpanElement, BadgeProps>((props, ref) => (
  <StyledBadge ref={ref} {...props} />
));

Badge.defaultProps = {
  dark: false,
  dot: false,
  invert: false,
  variant: 'primary',
};
