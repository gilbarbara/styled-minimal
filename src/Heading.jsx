import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import { px, spacer, themeGet } from './utils/helpers';
import { baseStyles, headingPropTypes } from './utils/system';

import Box from './Box';

const styles = (props: Object): string => {
  const { gutterBottom } = props;
  const headingWeight = themeGet(props, 'headingWeight');
  const headingSize = themeGet(props, 'headingSizes', { key: ['size', 'as'], base: 'h1' });

  return css`
    font-size: ${px(headingSize)};
    font-family: inherit;
    font-weight: ${headingWeight};
    line-height: ${baseStyles.lineHeight};
    margin: ${spacer(3)} 0 ${gutterBottom ? spacer(3) : 0};
    
    &:first-child {
      margin-top: 0;
    }
  `;
};

export const Heading = styled(Box)`
  ${styles};
`;

Heading.displayName = 'Heading';

Heading.propTypes = {
  as: headingPropTypes,
  children: PropTypes.node.isRequired,
  gutterBottom: PropTypes.bool,
  /** element size */
  size: headingPropTypes,
  ...Box.propTypes,
};

Heading.defaultProps = {
  as: 'h1',
};

export default Heading;
