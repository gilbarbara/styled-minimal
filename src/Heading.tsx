import * as React from 'react';
import styled, { css } from 'styled-components';

import { getTheme, px, spacer } from './utils/helpers';
import { baseStyles, sx, system } from './utils/system';

import { HeadingProps } from './types';

export const StyledHeading = styled('h1')<HeadingProps>(
  props => {
    const { as, level, gutter } = props;
    const headingWeight = getTheme(props, 'headingWeight');
    const marginBottom = gutter ? spacer(3) : 0;
    let headingSize = getTheme(props, 'headingSizes')[as!];

    if (level) {
      headingSize = getTheme(props, 'headingSizes')[`h${level}`];
    }

    return css`
      ${baseStyles};
      font-size: ${px(headingSize)};
      font-family: inherit;
      font-weight: ${headingWeight};
      line-height: 1;
      margin-bottom: ${marginBottom};
      margin-top: ${spacer(2)};

      &:first-child {
        margin-top: ${0};
      }
    `;
  },
  sx,
  system,
);

export const Heading = React.forwardRef<HTMLHeadingElement, HeadingProps>((props, ref) => (
  <StyledHeading ref={ref} {...props} />
));

Heading.defaultProps = {
  as: 'h1',
  gutter: false,
};
