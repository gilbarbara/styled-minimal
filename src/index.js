export { default as Alert } from './Alert';
export { default as Badge } from './Badge';
export { default as Box } from './Box';
export { default as Button } from './Button';
export { default as Code } from './Code';
export { default as Container } from './Container';
export { default as Embed } from './Embed';
export { default as Flex } from './Flex';
export { default as Fieldset } from './Fieldset';
export { default as Form } from './Form';
export { default as FormGroup } from './FormGroup';
export { default as Group } from './Group';
export { default as Heading } from './Heading';
export { default as Image } from './Image';
export { default as Input } from './Input';
export { default as Label } from './Label';
export { default as Legend } from './Legend';
export { default as Link } from './Link';
export { default as List } from './List';
export { default as Paragraph } from './Paragraph';
export { default as Screen } from './Screen';
export { default as Select } from './Select';
export { default as Switch } from './Switch';
export { default as Table } from './Table';
export { default as Text } from './Text';
export { default as Textarea } from './Textarea';
