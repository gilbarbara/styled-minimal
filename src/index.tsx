export { Alert } from './Alert';
export { Badge } from './Badge';
export { Box } from './Box';
export { Button } from './Button';
export { ButtonGroup } from './ButtonGroup';
export { Code } from './Code';
export { CodeBlock } from './CodeBlock';
export { Container } from './Container';
export { Embed } from './Embed';
export { Flex } from './Flex';
export { Fieldset } from './Fieldset';
export { Form } from './Form';
export { FormGroup } from './FormGroup';
export { Grid } from './Grid';
export { Group } from './Group';
export { Heading } from './Heading';
export { Image } from './Image';
export { Checkbox, Input, Radio } from './Input';
export { Label } from './Label';
export { Legend } from './Legend';
export { Link } from './Link';
export { DL, OL, UL } from './List';
export { Loader } from './Loader';
export { Paragraph } from './Paragraph';
export { RadioGroup } from './RadioGroup';
export { Select } from './Select';
export { Switch } from './Switch';
export { Table } from './Table';
export { Text } from './Text';
export { Textarea } from './Textarea';

export { default as system } from '@styled-system/css';
export { mergeTheme as getTheme, px, responsive } from './utils/helpers';
