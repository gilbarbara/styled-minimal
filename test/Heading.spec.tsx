import * as React from 'react';
import { render } from '@testing-library/react';

import { Heading } from '../src/Heading';

describe('Heading', () => {
  it('should render properly', () => {
    const { container } = render(<Heading>Far far away</Heading>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('with `gutter`', () => {
    const { container } = render(<Heading gutter>Far far away</Heading>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('with `as`', () => {
    const { container } = render(<Heading as="h4">Far far away</Heading>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('with `level`', () => {
    const { container } = render(<Heading level={4}>Far far away</Heading>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
