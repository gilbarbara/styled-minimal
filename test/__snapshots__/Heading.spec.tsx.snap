// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  font-size: 48px;
  font-family: inherit;
  font-weight: 700;
  line-height: 1;
  margin-bottom: 0;
  margin-top: 8px;
}

.c0:first-child {
  margin-top: 0;
}

<h1
  class="c0"
>
  Far far away
</h1>
`;

exports[`Heading with \`as\` 1`] = `
.c0 {
  box-sizing: border-box;
  font-size: 26px;
  font-family: inherit;
  font-weight: 700;
  line-height: 1;
  margin-bottom: 0;
  margin-top: 8px;
}

.c0:first-child {
  margin-top: 0;
}

<h4
  class="c0"
>
  Far far away
</h4>
`;

exports[`Heading with \`gutter\` 1`] = `
.c0 {
  box-sizing: border-box;
  font-size: 48px;
  font-family: inherit;
  font-weight: 700;
  line-height: 1;
  margin-bottom: 16px;
  margin-top: 8px;
}

.c0:first-child {
  margin-top: 0;
}

<h1
  class="c0"
>
  Far far away
</h1>
`;

exports[`Heading with \`level\` 1`] = `
.c0 {
  box-sizing: border-box;
  font-size: 26px;
  font-family: inherit;
  font-weight: 700;
  line-height: 1;
  margin-bottom: 0;
  margin-top: 8px;
}

.c0:first-child {
  margin-top: 0;
}

<h1
  class="c0"
>
  Far far away
</h1>
`;
